ethexe/common/src/crypto/address.rs

```diff
@@ -133,6 +134,82 @@ impl From<Address> for alloy_primitives::Address {
     }
 }
 
+/// [`ValidatorsVec`] is a wrapper over non-empty vector of [`Address`].
+/// It is needed because `NonEmpty` does not implement `Encode` and `Decode`.
+#[derive(
+    Debug,
+    Clone,
+    Default,
+    PartialEq,
+    Eq,
+    Hash,
+    derive_more::Deref,
+    derive_more::DerefMut,
+    derive_more::IntoIterator,
+)]
+pub struct ValidatorsVec(NonEmpty<Address>);
+
+// parity-scale-codec Encode / Decode implementations
+impl Encode for ValidatorsVec {
+    fn encode(&self) -> Vec<u8> {
+        Into::<Vec<_>>::into(self.0.clone()).encode()
+    }
+}
+
+impl Decode for ValidatorsVec {
+    fn decode<I: parity_scale_codec::Input>(
+        input: &mut I,
+    ) -> Result<Self, parity_scale_codec::Error> {
+        let inner: Vec<Address> = Decode::decode(input)?;
+        NonEmpty::from_vec(inner)
+            .map(Self)
+            .ok_or(parity_scale_codec::Error::from(
+                "Failed to decode ValidatorsVec: empty vector",
+            ))
+    }
+}
+
+#[derive(Debug, Display, Error)]
+#[display("{:?}", self)]
+#[debug("ValidatorsVec cannot be created from empty vector")]
+pub struct TryFromVecError;
```

Comment:
EmptyvalidatorsError could be more suitable naming to use this error in other constructors also